src/Microsoft.ML.CodeGen/CodeGenerator/CSharp/CodeGeneratorSettings.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using Microsoft.ML.AutoML;
+
+namespace Microsoft.ML.CodeGenerator.CSharp
+{
+    public class CodeGeneratorSettings
+    {
+        public CodeGeneratorSettings()
+        {
+            // set default value
+            Target = GenerateTarget.Cli;
+        }
+
+        public string LabelName { get; set; }
+
+        public string ModelPath { get; set; }
+
+        public string OutputName { get; set; }
+
+        public string OutputBaseDir { get; set; }
+
+        public string TrainDataset { get; set; }
+
+        public string TestDataset { get; set; }
+
+        public GenerateTarget Target { get; set; }
+
+        internal TaskKind MlTask { get; set; }
+
+    }
+
+    public enum GenerateTarget
```

Comment:
This seem to be used only to print the right annotation for the platform. I might recommend just having a string field here for the annotation text.

This would address my other comment: https://github.com/dotnet/machinelearning/pull/4043/files#r317240647